modules/core/src/main/java/org/apache/ignite/internal/client/router/impl/GridRouterCommandLineStartup.java

```diff
@@ -49,20 +50,21 @@
      *
      * @param beans Beans loaded from spring configuration file.
      */
-    public void start(Map<Class<?>, Object> beans) {
-        log = (IgniteLogger)beans.get(IgniteLogger.class);
-
-        if (log == null) {
+    public void start(Map<Class<?>, Collection> beans) {
+        if (F.isEmpty(beans.get(IgniteLogger.class))) {
```

Comment:
Let's just change 
`log = (IgniteLogger)beans.get(IgniteLogger.class);`
to
`log = F.<IgniteLogger>first(beans.get(IgniteLogger.class));`

And do the same with tcpCfg.

Reply:
Fixed